Source/WebInspectorUI/UserInterface/Views/SpreadsheetCSSStyleDeclarationSection.js

```diff
@@ -694,6 +719,84 @@ WI.SpreadsheetCSSStyleDeclarationSection = class SpreadsheetCSSStyleDeclarationS
 
         this.dispatchEventToListeners(WI.SpreadsheetCSSStyleDeclarationSection.Event.FilterApplied, {matches});
     }
+
+    _handleSpreadsheetSelectorFieldDidCommit(changed)
+    {
+        let selectorText = this._selectorElement.textContent.trim();
+        if (selectorText && changed) {
+            this.dispatchEventToListeners(WI.SpreadsheetCSSStyleDeclarationSection.Event.SelectorWillChange);
+            this._style.ownerRule.setSelectorText(selectorText).finally(this._renderSelector.bind(this));
+        } else
+            this._discardSelectorChange();
+    }
+
+    _handleSpreadsheetSelectorFieldWillNavigate(direction)
+    {
+        if (direction === "forward")
+            this._propertiesEditor.startEditingFirstProperty();
+        else if (direction === "backward") {
+            if (this._delegate.spreadsheetCSSStyleDeclarationSectionStartEditingAdjacentRule) {
+                const delta = -1;
+                this._delegate.spreadsheetCSSStyleDeclarationSectionStartEditingAdjacentRule(this, delta);
+            } else
+                this._propertiesEditor.startEditingLastProperty();
+        }
+    }
+
+    _handleSpreadsheetGroupingFieldDidCommit(textField, changed)
+    {
+        let groupingTextElement = textField.element;
+        let text = groupingTextElement.textContent.trim();
+
+        if (!changed || !text) {
+            this._handleSpreadsheetGroupingFieldDidDiscard(textField);
+            return;
+        }
+
+        this.dispatchEventToListeners(WI.SpreadsheetCSSStyleDeclarationSection.Event.SelectorWillChange);
```

Comment:
should we create a `GroupingWillChange` event so we're not overloading this one?